File: eFrizer/eFrizer.Model/ApplicationUser/ApplicationUser.cs


using System.Collections.Generic;

namespace eFrizer.Model
{
    public partial class ApplicationUser
    {
        //TODO: Create search and put requests for this 
        public int ApplicationUserId { get; set; }
        public string Name { get; set; }
        public string Surname { get; set; }
        public string Username { get; set; }
        public string Description { get; set; }
        //TODO: Why is this, on a GET Application Users Request, showing only the role of the currently logged in user but not other user roles?
        //When the authentication is disabled, this does not happen. What does the BasicAuthHandler do that triggers this type of behaviour?
        public virtual ICollection<ApplicationUserRole> ApplicationUserRoles { get; set; }
    }
}

File: eFrizer/flutter_efrizer/lib/models/application_user/application_user.dart
class ApplicationUser {
  final int applicationUserId;
  final String? name;
  final String? surname;
  final String? description;
  final String? username;
  final List<dynamic> roles;

  ApplicationUser(
      {required this.applicationUserId,
      this.name,
      this.surname,
      required this.description,
      required this.username,
      required this.roles});

  void SetParameter(String Username, String Password) {}

  factory ApplicationUser.fromJson(Map<String, dynamic> json) {
    return ApplicationUser(
        applicationUserId: int.parse(json["applicationUserId"].toString()),
        name: json["name"],
        surname: json["surname"],
        description: json["description"],
        username: json["username"],
        roles: json["applicationUserRoles"] as List<dynamic>);
  }

  Map<String, dynamic> toJson() => {
        "username": username,
        "name": name,
        "surname": surname,
        "description": description,
      };

  Map<String, dynamic> toJsonWithId() => {
        "applicationUserId": applicationUserId,
        "username": username,
        "name": name,
        "surname": surname,
        "description": description,
        "roles": roles
      };
}
